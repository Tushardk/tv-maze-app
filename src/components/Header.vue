<script setup lang="ts">
import HeaderLogo from '@/assets/tvm-header-logo.png';
import SearchBar from './SearchBar.vue';
</script>

<template>
	<header class="bg-stone-700 p-4">
		<div class="md:flex justify-between items-center">
			<!-- App Logo-->
			<div class="mb-4 md:m-auto md:basis-1/5">
				<router-link :to="{ name: 'home' }">
					<img :src="HeaderLogo" class="m-auto">
				</router-link>
			</div>
			<!-- App search bar -->
			<div class="mb-4 md:m-auto md:basis-4/5 md:p-4">
				<SearchBar />
			</div>
			<!-- App login & register buttons -->
			<div class="flex justify-between items-center md:basis-1/5 sm:justify-end">
				<div>
					<button class="bg-teal-700 hover:bg-teal-800 text-white font-bold py-2 px-4 rounded sm:mr-2">
						Register
					</button>
				</div>
				<div>
					<button class="bg-teal-700 hover:bg-teal-800 text-white font-bold py-2 px-4 rounded">
						Login
					</button>
				</div>
			</div>
		</div>
	</header>
</template>	