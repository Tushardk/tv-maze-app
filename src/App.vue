<script setup lang="ts">
import Header from './components/AppHeader.vue';
</script>

<template>
  <Header />
  <RouterView />
</template>
