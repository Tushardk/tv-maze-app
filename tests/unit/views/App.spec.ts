import { shallowMount } from '@vue/test-utils';
import App from '@/App.vue';
import Header from '@/components/AppHeader.vue';

describe('Testing HomeView.vue wrapper component', () => {
  const shallowWrapper = shallowMount(App, {
    stubs: ['router-view'],
  });
  it('App.vue should be rendered correctly', () => {
    expect(shallowWrapper.element).toMatchSnapshot();
  });

  it('child component PbdbMasterLayout should be present', () => {
    expect(shallowWrapper.findComponent(Header).exists()).toBe(true);
  });
});
