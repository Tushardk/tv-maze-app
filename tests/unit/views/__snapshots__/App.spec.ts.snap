// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing HomeView.vue wrapper component App.vue should be rendered correctly 1`] = `
<div
  data-v-app=""
>
  
  <header-stub />
  <routerview />
  
</div>
`;
